### PR TITLE
ci: add cilium identities check to nightly pipeline

### PR DESCRIPTION
**Pull request by camrynl**
**Reason for Change**:

- This PR updates cilium operator arguments for cilium identities to lower the garbage collection interval. 
- Adding a stage in the nightly pipeline to verify that cilium identities are removed once the cilium endpoint is deleted. 

This check will only run for the nightly pipeline and will be skipped in the PR runs. It was previously added to the old version of our nightly pipeline.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

File: .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml [hunk 1]
```diff
@@ -166,6 +170,33 @@ steps:
...
+          fi
+          sleep 2s
+        done
+        sleep 20s
+        echo "Verify cilium identities are deleted"
+        checkIdentity="$(kubectl get ciliumidentity -o json | grep cilium-test | jq -e 'length == 0')"
+        if [[ -n $checkIdentity ]]; then
+          echo "##[error]Cilium Identities still present"
+        else
+          printf -- "Identities deleted\n"
```
Comment (jpayne3506) [inline .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml:192]: Can we have additional logging here? If identities are still present -> see what they are and confirm which identities are being deleted (if possible).

File: .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml [hunk 2]
```diff
@@ -166,6 +170,33 @@ steps:
...
+        fi 
+      else
+        echo "skip cilium identities check for PR pipeline"
+      fi
+    name: "CiliumIdentites"
```
[the remark this answers is not in the document]
Comment (camrynl) [reply .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml]: thanks, fixed

File: .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml (hunk 1 above)
Comment (camrynl) [reply .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml]: Sure, the check is verifying that the identities are being deleted from the cilium-test ns. I updated the logging with references to the namespace.

File: .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml [hunk 3]
```diff
@@ -166,6 +171,33 @@ steps:
...
+        ns="cilium-test"
+        while true; do
+          pods=$(kubectl get pods -n $ns --no-headers=true 2>/dev/null)
+          if [[ -z "$pods" ]]; then
+            echo "No pods found"
+              break
+          fi
+          sleep 2s
+        done
+        sleep 20s
```
Comment (vipul-21) [inline .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml:187]: Why are sleeping again after checking the deletion of pods ? For cilium `gc` ?

File: .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml [hunk 4]
```diff
@@ -166,6 +171,33 @@ steps:
...
+        echo "Verify cilium identities are deleted from cilium-test"
+        checkIdentity="$(kubectl get ciliumidentity -o json | grep cilium-test | jq -e 'length == 0')"
+        if [[ -n $checkIdentity ]]; then
+          echo "##[error]Cilium Identities still present in cilium-test namespace"
+        else
+          printf -- "Identities deleted from cilium-test namespace\n"
```
Comment (vipul-21) [inline .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml:193]: Any reason we added `printf` and not `echo`

File: .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml (hunk 3 above)
Comment (camrynl) [reply .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml:187]: Yes, need to wait for cilium gc

File: .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml (hunk 4 above)
Comment (camrynl) [reply .pipelines/singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml:193]: When I wrote this for the original nightly pipeline I recall I had to change the first case to echo with azp syntax in order to indicate actual failure in the pipeline.

**Review by jpayne3506 — APPROVED** [review]

**Pull request merged**